modules/reboot/src/main/java/org/mule/runtime/module/reboot/internal/MuleContainerWrapperProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.reboot.internal;
+
+import static java.lang.String.format;
+import static java.lang.System.getProperty;
+
+/**
+ * Helps with the creation and provisioning of the {@link MuleContainerWrapper} implementation instance as a singleton.
+ * <p>
+ * Only thread-safe for threads started after the first provisioning.
+ *
+ * @since 4.5
+ */
+public class MuleContainerWrapperProvider {
+
+  private static final String MULE_BOOTSTRAP_CONTAINER_WRAPPER_CLASS_SYSTEM_PROPERTY = "mule.bootstrap.container.wrapper.class";
+  private static final String DEFAULT_WRAPPER_IMPL_CLASS = "org.mule.runtime.module.tanuki.internal.MuleContainerTanukiWrapper";
```

Comment:
we should make sure the launcher scripts provide this instead of having this default and loose coupling to tanuki here